src/main/js/utils/generate.js

```diff
@@ -76,20 +82,19 @@ const generateSchedules = (values, loadQuery = '') => (dispatch, getState) => {
           if (!gradeMap[name]) return 0;
 
           if (!gradeMap[name][instructor1] && !gradeMap[name][instructor2]) return 0;
-          if (!gradeMap[name][instructor1]) return 1;
-          if (!gradeMap[name][instructor2]) return -1;
+          if (!gradeMap[name][instructor1]) return gradeMap[name][instructor2] - gradeMap[name].Staff;
+          if (!gradeMap[name][instructor2]) return gradeMap[name].Staff - gradeMap[name].Staff;
```

Comment:
Didn't refactor using `AVERAGE`. Also try to pare down these if statements like my other suggestions.

Reply:
Might need help refactoring, but fixed.